pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go

```diff
@@ -26,13 +26,21 @@ type CloudStackDatacenterConfigSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
+	// FailureDomains is a list of one or more failure domains
+	FailureDomains []CloudStackFailureDomain `json:"failureDomains"`
+}
+
+// CloudStackFailureDomain defines a CAPI failure domain which serves to implements HA and improve availability by distributing machines across "failure domains"
+type CloudStackFailureDomain struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
 	// Domain contains a grouping of accounts. Domains usually contain multiple accounts that have some logical relationship to each other and a set of delegated administrators with some authority over the domain and its subdomains
 	// This field is considered as a fully qualified domain name which is the same as the domain path without "ROOT/" prefix. For example, if "foo" is specified then a domain with "ROOT/foo" domain path is picked.
 	// The value "ROOT" is a special case that points to "the" ROOT domain of the CloudStack. That is, a domain with a path "ROOT/ROOT" is not allowed.
-	//
 	Domain string `json:"domain"`
-	// Zones is a list of one or more zones that are managed by a single CloudStack management endpoint.
-	Zones []CloudStackZone `json:"zones"`
+	// Zone is a CloudStackZone that is managed by a single CloudStack management endpoint.
+	Zone CloudStackZone `json:"zone"`
```

Comment:
We should also include a Name field here to match it up with the ini config file and ensure that all the AvailabilityZones match up with the cmk profiles